src/digraph.rs

```diff
@@ -2720,6 +2720,56 @@ impl PyDiGraph {
         self.clone()
     }
 
+    /// Reverse the direction of all edges in the graph, in place.
+    ///
+    /// This method modifies the graph instance to reverse the direction of all edges.
+    /// It does so by iterating over all edges in the graph and removing each edge,
+    /// then adding a new edge in the opposite direction with the same weight.
+    ///
+    /// # Arguments
+    ///
+    /// * `py`: A reference to the Python interpreter.
+    ///
+    /// # Returns
+    ///
+    /// None.
+    ///
+    /// # Panics
+    ///
+    /// This method will panic if the edge indices or weights are not valid.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rustworkx::{DiGraph, WeightedGraph};
+    ///
+    /// let mut graph = DiGraph::<(), i32>::new();
+    /// graph.add_edge(0, 1, 3).unwrap();
+    /// graph.add_edge(1, 2, 5).unwrap();
+    /// graph.add_edge(2, 3, 2).unwrap();
+    ///
+    /// graph.reverse_inplace(py);
+    ///
+    /// assert_eq!(graph.edges().collect::<Vec<_>>(), vec![(3, 2), (2, 1), (1, 0)]);
+    /// ```
```

Comment:
One of the weird quirks of rustworkx is that docstrings for these functions are actually built using sphinx and not rustdoc. So while this is a valid rust docstring to document this function it's actually going to have issues when sphinx tries to parse it. Mostly because sphinx uses restructured text instead of markdown but also some of the directives are correct. I'd rewrite this as:

```suggestion
    ///  .. note:: The edge indices will change if this method is used to reverse the edges
    ///
    /// For example::
    ///
    ///     import rustworkx as rx
    ///
    ///     graph = rx.PyDiGraph()
    ///     graph.add_nodes_from(range(4))
    ///     graph.add_edge(0, 1, 3)
    ///     graph.add_edge(1, 2, 5)
    ///     graph.add_edge(2, 3, 2)
    ///
    ///     graph.reverse_inplace()
    ///
    ///     assert graph.edge_list() == [(3, 2), (2, 1), (1, 0)]
```

I removed the panic detail because I don't think the function can ever panic in practice.